### PR TITLE
Clarify what values `BorrowedHandle`, `OwnedHandle` etc. can hold.

### PR DESCRIPTION
**Pull request by sunfishcode**
Reword the documentation to clarify that when `BorrowedHandle`, `OwnedHandle`, or `HandleOrNull` hold the value `-1`, it always means the current process handle, and not `INVALID_HANDLE_VALUE`.

`-1` should only mean `INVALID_HANDLE_VALUE` after a call to a function documented to return that to report errors, which should lead I/O functions to produce errors rather than succeeding and producing `OwnedHandle` or `BorrowedHandle` values. So if a consumer of an `OwnedHandle` or `BorrowedHandle` ever sees them holding a `-1`, it should always mean the current process handle.

### CODE REVIEW

File: library/std/src/os/windows/io/handle.rs [hunk 1]
```diff
@@ -75,11 +77,11 @@ pub struct OwnedHandle {
 /// `NULL`. This ensures that such FFI calls cannot start using the handle without
 /// checking for `NULL` first.
 ///
-/// This type considers any value other than `NULL` to be valid, including `INVALID_HANDLE_VALUE`.
-/// This is because APIs that use `NULL` as their sentry value don't treat `INVALID_HANDLE_VALUE`
-/// as special.
+/// This type may hold any handle value that [`OwnedHandle`] may hold, except `NULL`. It may
```
Comment (kennytm) [inline library/std/src/os/windows/io/handle.rs:80]: This sentence sounds like this type cannot hold NULL, but it's the other way around (the type can hold NULL as the sentry value).

**Comment by joshtriplett** [comment]
Seems reasonable. Are there *no* other circumstances in which such a handle might ever hold -1? Is there no way to end up with one yourself, if constructed that way rather than returned from a Windows API?

**Comment by sunfishcode** [comment]
> Seems reasonable. Are there _no_ other circumstances in which such a handle might ever hold -1? Is there no way to end up with one yourself, if constructed that way rather than returned from a Windows API?

The value of -1 is documented [here], though it does also say that in theory the current process handle could have a different value in future OS versions. I've now added a patch to change the wording to reflect that `OwnedHandle` etc. interpret `-1` as a valid handle, with the current process handle being an example of such a valid handle.

[here]: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentprocess#remarks

**Comment by ChrisDenton** [comment]
I'd only add that all valid handle values should originally come from the Windows API in one way or another. Like pointers they shouldn't be created out of thin air. This goes for pseudo-handles too. As mentioned above, even though they may have a known value this value isn't actually guaranteed.

**Pull request merged**